scripts/main.js

```diff
@@ -0,0 +1,348 @@
+"use strict";
+document.addEventListener("keydown", (event) => move(event.key));
+
+const rows = document.querySelectorAll("tr");
+
+const button = document.querySelector(".button");
+const score = document.querySelector(".game-score");
+
+let scoreVal = 0;
+
+const board = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+
+const message_start = document.querySelector(".message-start");
+const message_lose = document.querySelector(".message-lose");
+const message_win = document.querySelector(".message-win");
+
+button.addEventListener("click", () => {
+  if (button.classList.contains("start")) {
+    start();
+  } else {
+    restart();
+  }
+});
+
+function start() {
+  button.innerHTML = "Restart";
+  button.classList.add("restart");
+  button.classList.remove("start");
+
+  message_start.classList.add("hidden");
+
+  newCell();
+  newCell();
+  drawBoard();
+}
```

Comment:
The 'start' function directly manipulates the DOM to change the button's text and classes. This can be refactored into a separate function that handles the button state, improving code reusability and separation of concerns.